userspace/libscap/scap.c

```diff
@@ -445,6 +445,76 @@ scap_t* scap_open_udig_int(char *error, int32_t *rc,
 }
 #endif // !defined(HAS_CAPTURE) || defined(CYGWING_AGENT)
 
+scap_t* scap_open_test_input_int(char *error, int32_t *rc, scap_open_args *args)
+{
+	scap_t* handle = NULL;
+
+	//
+	// Allocate the handle
+	//
+	handle = (scap_t*) calloc(1, sizeof(scap_t));
+	if(!handle)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the scap_t structure");
+		*rc = SCAP_FAILURE;
+		return NULL;
+	}
+
+	handle->m_vtable = &scap_test_input_engine;
+	handle->m_engine.m_handle = handle->m_vtable->alloc_handle(handle, handle->m_lasterr);
+	if(!handle->m_engine.m_handle)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "error allocating the engine structure");
+		free(handle);
+		return NULL;
+	}
+
+	*rc = handle->m_vtable->init(handle, args);
+	if(*rc != SCAP_SUCCESS)
+	{
+		snprintf(error, SCAP_LASTERR_SIZE, "%s", handle->m_lasterr);
+		scap_close(handle);
+		return NULL;
+	}
+
+	//
+	// Preliminary initializations
+	//
+	handle->m_mode = SCAP_MODE_LIVE;
+
+	handle->m_ncpus = 1;
+
+	handle->m_fake_kernel_proc.tid = -1;
+	handle->m_fake_kernel_proc.pid = -1;
+	handle->m_fake_kernel_proc.flags = 0;
+	snprintf(handle->m_fake_kernel_proc.comm, SCAP_MAX_PATH_SIZE, "kernel");
+	snprintf(handle->m_fake_kernel_proc.exe, SCAP_MAX_PATH_SIZE, "kernel");
+	handle->m_fake_kernel_proc.args[0] = 0;
+	handle->refresh_proc_table_when_saving = true;
+
+	handle->m_suppressed_comms = NULL;
+	handle->m_num_suppressed_comms = 0;
+	handle->m_suppressed_tids = NULL;
+	handle->m_num_suppressed_evts = 0;
+
+	handle->m_proc_callback = args->proc_callback;
+	handle->m_proc_callback_context = args->proc_callback_context;
+
+	if ((*rc = copy_comms(handle, args->suppressed_comms)) != SCAP_SUCCESS)
+	{
+		scap_close(handle);
+		snprintf(error, SCAP_LASTERR_SIZE, "error copying suppressed comms");
+		return NULL;
+	}
+
+	if(handle->m_vtable->start_capture(handle->m_engine) != SCAP_SUCCESS)
+	{
+		scap_close(handle);
+		return NULL;
+	}
```

Comment:
Do we really need this? the `noop_start_capture` always return `SCAP_SUCCESS` without doing anything else :thinking:

Reply:
I don't want to speak for Luca, but personally I have a "better safe than sorry" policy, granted this is just for testing but if the `start_capture` function for this engine is changed to something that can fail, it could lead to some pretty erratic error messages. Just my 2 cents 🤷🏻‍♂️

Reply:
Yes, let's keep it. Somewhere down my patch queue I have one unifying all the various open methods, so the more similar they are, the less ugly it's going to be :)